sloution.kts

```diff
@@ -0,0 +1,82 @@
+fun main() {
+    val shape1 = Shape()  //creating object shape1 from class Shape
+    val tri1 = Triangle() //creating object tri1 from class Triangle
+    val rec1 = Rectangle() //creating object rec1 from class Rectangle
+    val cir1 = Circle() //creating object cir1 from class Circle
+
+    println(tri1.info())  //print info function for triangle
+    println(rec1.info())  //print info function for rectangle
+    println(cir1.info())  //print info function for circle
+
+    println("The Area of Triangle = " + tri1.computeArea(4.0, 4.0)) //send values to computeArea function for triangle
+    println("The Area of Rectangle = " + rec1.computeArea(4.0, 4.0)) //send values to computeArea function for Rectangle
+    println("The Area of Circle = " + cir1.computeArea(4.0)) //send values to computeArea function for Circle
+}
+
+
+open class Shape {     //defining super Class
+
+    open val shapeName = ""  // variable for shape name
+    open val shapeColor = ""  // variable for shape name
+    open var result: Double = 0.0  // variable for area result
+    fun info(): String {  // defining info function
+        return "Shape Name: $shapeName\nShape Color: $shapeColor"  //return the name and color for shape
+    }
+
+    open fun computeArea(): Double { // defining computeArea Function
+        val num1: Double = 0.0  // variable for num1
+        val num2: Double = 0.0  // variable for num2
+        var result: Double = 0.0  // variable for area result
+        return result  // return the area result
+    }
```

Comment:
why did you make it open explain what this code dose ?
don't restate the sentences

Reply:
this shape function is open so we can use inheritance and the function have 3 variable for shape name , color , result
open variable used to change the values for name and color val variable , for the result i was just trying to check if its will work

info function will return the shape name and shape color for the class when its called from other class
compute area will compute area when its called form other class